File: ignisign-ios/Public/Signatures.swift
//
//  Signatures.swift
//  ignisign-ios
//
//  Created by Marc Nigdélian on 19/12/2023.
//

import Foundation

enum IgnisignSignatureMode {
    case INIT
    case SIGNED
    case FAILED
}

enum IgnisignSignatureStatus {
    case INIT
    case SIGNED
    case FAILED
}

enum IgnisignIntegrationMode {
    case BY_SIDE
    case EMBEDDED
}

struct IgnisignSignature {
    var _id: String
    var appId: String
    var appEnv: IgnisignApplicationEnv
    var signerId: String
    var signerKeyId: String
    var sessionId: String
    var documentId: String
    var status: IgnisignSignatureStatus
    var mode: IgnisignIntegrationMode
    var ocspCheckValue: Any
    var contentHash: String
    var signature: String
    var signatureValue: String
    var signatureProperties: String
    var signedPropertiesHash: String
    var signingIp: String
    var signingTime: String
    var certificate: String
}

struct IgnisignApplicationSignatureMetadata {
    var appName: String
    var logoB64: String
    var logoDarkB64: String
    var rootUrl: String
    var primaryColor: IgnisignApplicationVariationColor
    var secondaryColor: IgnisignApplicationVariationColor
}

struct IgnisignSignatureImagesDto {
    let documentId: String
    let signatures: [SignatureImage]

    struct SignatureImage {
        let signerId: String
        let imgB64: String
    }
}

File: ignisign-android/ignisign/src/main/java/com/ignisign/ignisign/public/Signatures.kt
package com.ignisign.ignisign.public

import com.ignisign.ignisign.IgnisignApplicationEnv
import com.ignisign.ignisign.IgnisignApplicationVariationColor

enum class IgnisignSignatureMode {
    SERVER_SIDE_SIGNATURE,
    CLIENT_SIDE_SIGNATURE
}

enum class IgnisignSignatureStatus {
    INIT,
    SIGNED,
    FAILED
}

enum class IgnisignIntegrationMode {
    BY_SIDE,
    EMBEDDED
}

data class IgnisignSignature(
    val _id: String? = null,
    val appId: String,
    val appEnv: IgnisignApplicationEnv,
    val signerId: String,
    val signerKeyId: String,
    val sessionId: String? = null,
    val documentId: String,
    val status: IgnisignSignatureStatus,
    val mode: IgnisignSignatureMode,
    val ocspCheckValue: Any? = null,
    val contentHash: String? = null,
    val signature: String? = null,
    val signatureValue: String? = null,
    val signedProperties: String? = null,
    val signedPropertiesHash: String? = null,
    val signingIp: String? = null,
    val signingTime: String? = null,
    val certificate: String? = null
)

data class IgnisignApplicationSignatureMetadata(
    val appName: String,
    val logoB64: String? = null,
    val logoDarkB64: String? = null,
    val rootUrl: String? = null,
    val primaryColor: IgnisignApplicationVariationColor? = null,
    val secondaryColor: IgnisignApplicationVariationColor? = null
)

data class IgnisignSignatureImagesDto(
    val documentId: String,
    val signatures: List<SignatureImage>
) {
    data class SignatureImage(
        val signerId: String,
        val imgB64: String
    )
}

data class IgnisignDocumentSignatureProofCustomizationDto(
    val html: String
)
